tensorrt_llm/_torch/pyexecutor/py_executor.py

```diff
@@ -368,7 +374,12 @@ def cancel_request(self, id: int):
         Args:
             id (int): The request id for which to cancel the response
         """
-        self.canceled_req_ids.insert(id)
+        try:
+            self.enqueue_lock.acquire()
```

Comment:
Hi @Shunkangz, I unifiy the logic of canceled requsts and normal requests. All requests will be fetched from request queue now.